File: BackEnd/Core/Features/Product/Queries/Results/ProductResponse.cs
namespace Core.Features.Product.Queries.Results
{
    public class ProductResponse
    {
        public Guid Id { get; set; }
        public Guid OrganizationId { get; set; }

        public string Name { get; set; } = null!;
        public string? Description { get; set; }
        public List<string> Images { get; set; } = new List<string>();
        public string? MainImageUrl { get; set; }
        public int StockQuantity { get; set; }

        public bool IsAvailableForDirectSale { get; set; }
        public decimal DirectSalePrice { get; set; }

        public bool IsAvailableForAuction { get; set; }
        public decimal StartBiddingPrice { get; set; }

        public string? Barcode { get; set; }
        public DateTime? ExpiryDate { get; set; }

        public string? WeightUnit { get; set; }
        public string? ProductType { get; set; }
    }
}

File: FrontEnd/src/app/core/model/Organization/productResponse.ts
export interface ProductResponse {
    id: string;
    organizationId: string;
    name: string;
    description?: string;
    images: string[];
    mainImageUrl?: string;
    stockQuantity: number;
    isAvailableForDirectSale: boolean;
    directSalePrice: number;
    isAvailableForAuction: boolean;
    startBiddingPrice: number;
    barcode?: string;
    expiryDate?: string;
    weightUnit?: string;
    productType?: string;
}
